app/code/Magento/Swatches/Test/Mftf/Test/StorefrontRedirectToFirstPageOnFilteringBySwatchTest.xml

```diff
@@ -0,0 +1,103 @@
+            <magentoCLI command="config:set catalog/frontend/grid_per_page_values 12,24,36" stepKey="setDefaultGridPerPage"/>
+            <magentoCLI command="cache:clean config full_page" stepKey="cleanInvalidatedCaches"/>
+
+            <deleteData createDataKey="createCategory" stepKey="deleteCategory"/>
+            <deleteData createDataKey="createSimpleProduct1" stepKey="deleteSimpleProduct1"/>
+            <deleteData createDataKey="createSimpleProduct2" stepKey="deleteSimpleProduct2"/>
+            <deleteData createDataKey="createSimpleProduct3" stepKey="deleteSimpleProduct3"/>
+        </after>
+
+        <amOnPage url="{{AdminProductAttributeSetEditPage.url}}/{{AddToDefaultSet.attributeSetId}}/" stepKey="onAttributeSetEdit"/>
+        <actionGroup ref="AssignAttributeToGroupActionGroup" stepKey="assignAttributeToGroup">
+            <argument name="group" value="Product Details"/>
+            <argument name="attribute" value="{{textSwatchAttribute.attribute_code}}"/>
+        </actionGroup>
+        <actionGroup ref="SaveAttributeSetActionGroup" stepKey="SaveAttributeSet"/>
+
+        <actionGroup ref="AdminOpenProductIndexPageActionGroup" stepKey="goToProductsGridPage"/>
+
+        <actionGroup ref="OpenEditProductOnBackendActionGroup" stepKey="goToProduct1EditPage">
+            <argument name="product" value="$$createSimpleProduct1$$"/>
+        </actionGroup>
+        <selectOption selector="{{AdminProductAttributesSection.attributeDropdownByCode(textSwatchAttribute.attribute_code)}}" userInput="textSwatchOption1" stepKey="selectProduct1AttributeOption"/>
+        <actionGroup ref="SaveProductFormActionGroup" stepKey="saveProduct1"/>
+
+        <actionGroup ref="AdminOpenProductIndexPageActionGroup" stepKey="goToProductsGridPage2"/>
+        <actionGroup ref="OpenEditProductOnBackendActionGroup" stepKey="goToProduct2EditPage">
+            <argument name="product" value="$$createSimpleProduct2$$"/>
+        </actionGroup>
+        <selectOption selector="{{AdminProductAttributesSection.attributeDropdownByCode(textSwatchAttribute.attribute_code)}}" userInput="textSwatchOption1" stepKey="selectProduct2AttributeOption"/>
+        <actionGroup ref="SaveProductFormActionGroup" stepKey="saveProduct2"/>
+
+        <actionGroup ref="AdminOpenProductIndexPageActionGroup" stepKey="goToProductsGridPage3"/>
+        <actionGroup ref="OpenEditProductOnBackendActionGroup" stepKey="goToProduct3EditPage">
+            <argument name="product" value="$$createSimpleProduct3$$"/>
+        </actionGroup>
+        <selectOption selector="{{AdminProductAttributesSection.attributeDropdownByCode(textSwatchAttribute.attribute_code)}}" userInput="textSwatchOption2" stepKey="selectProduct3AttributeOption"/>
+        <actionGroup ref="SaveProductFormActionGroup" stepKey="saveProduct3"/>
+
+        <magentoCLI command="cron:run --group=index" stepKey="runCronIndexer"/>
+
+        <amOnPage url="$$createCategory.name$$.html" stepKey="amOnCategoryPage"/>
```

Comment:
There's an Action Group that uses `url_key`, not `name` of category.
I can look for it tomorrow.